src/persist/src/indexed/columnar.rs

```diff
@@ -511,8 +511,10 @@ impl Default for ColumnarRecordsVecBuilder {
 }
 
 impl ColumnarRecordsVecBuilder {
-    #[cfg(test)]
-    fn new_with_len(key_val_data_max_len: usize) -> Self {
+    /// Create a new ColumnarRecordsVecBuilder with a specified max size.
+    ///
+    /// Only used for testing.
+    pub(crate) fn new_with_len(key_val_data_max_len: usize) -> Self {
```

Comment:
why'd we lose the #[cfg(test)]?

Reply:
we need to use this function in `compact_trace_blocking` in the case when the `key_val_max_len` passed to `Maintainer` is `Some(_)`. `compact_trace_blocking` is not cfg(test), and so the compiler complains if this function is, even though the key_val_max_len is only `Some(_)` in `Maintainer::new_for_testing` which is `cfg(test)` but the compiler can't infer that.

Reply:
I couldn't think of any nicer workarounds, curious if you have any ideas!